lib/ip_util.c

```diff
@@ -149,6 +149,55 @@ int rc_get_srcaddr(struct sockaddr *lia, const struct sockaddr *ria)
 	return 0;
 }
 
+/** Parse bindaddr config parameter to retrieve the source address
+ *
+ * bindaddr parameter in config file can have values: * or <ipv4>,<ipv6>
+ *   "192.168.11.11,2001::2" or "*,2001::2" or "192.168.11.11,*" or "*" or "192.168.11.11"
+ *   * - means the IP packet can have any outbound interface IP. 
+ *
+ * @param bindaddr_str bindaddr parameter from config file.
+ * @param srctxtaddr out paramter which will have IPv4 or IPv6 source address
+ * @param addr_type AF family type to be retrieved.
+ */
+void rc_get_source_ip(char *bindaddr_str, char *srctxtaddr, unsigned short addr_type)
+{
+	char *p_save;
+	char *p_ptr = NULL;
+
+	if (bindaddr_str == NULL)
+	{
+		rc_log(LOG_ERR, "bind address is null");
+		return;
+	}
+
+	char *p_bindaddr = strdup(bindaddr_str);
+	if (addr_type == AF_INET)
+	{
+		p_ptr = strtok_r(p_bindaddr, "," ,&p_save);
+		if (p_ptr != NULL)
+		{
+			rc_log(LOG_DEBUG, "%s: return address %s", __FUNCTION__, p_ptr);
+			strcpy(srctxtaddr, p_ptr);
+		}
+	}
+
+	if (addr_type == AF_INET6)
+	{
+		char *pch = strrchr(p_bindaddr, ',');
+		if (pch != NULL)
+		{
+			rc_log(LOG_DEBUG, "%s : inet6 string %s ", __FUNCTION__, pch);
+			strcpy(srctxtaddr, pch+1);
+		}
+	}
+
```

Comment:
It may be simpler to use strtok_r as you are using, and then what to return based on what there is found in the buffer.

e.g.,
```
do {
   ptr = strtok_r(p_bindaddr, "," ,&p_save);
   if (inet_addr(ptr) != INADDR_NONE) /* ipv4 */ {
      if (addr_type == AF_INET) ...
   } else { /* probably ipv6 */
      if (addr_type == AF_INET) ...
   }
   p_bindaddr = NULL;
} while (...);